Sources/ClientRuntime/Networking/Http/Middlewares/HeaderMiddleware.swift

```diff
@@ -5,10 +5,14 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-public struct HeaderMiddleware<OperationStackInput: HeaderProvider, OperationStackOutput>: Middleware {
+public struct HeaderMiddleware<OperationStackInput, OperationStackOutput>: Middleware {
```

Comment:
This middleware no longer requires the input to conform to a protocol, and takes a closure at initialization to get the needed headers instead.